source/extensions/filters/http/ratelimit/ratelimit.cc

```diff
@@ -293,6 +293,19 @@ VhRateLimitOptions Filter::getVirtualHostRateLimitOption(const Router::RouteCons
   return vh_rate_limits_;
 }
 
+std::string Filter::getDomain() {
+  std::string domain = config_->domain();
+  const auto* specific_per_route_config =
+      Http::Utility::resolveMostSpecificPerFilterConfig<FilterConfigPerRoute>(callbacks_);
+  if (specific_per_route_config != nullptr) {
+    std::string per_route_domain = specific_per_route_config->domain();
+    if (!per_route_domain.empty()) {
+      domain = per_route_domain;
+    }
+  }
+  return domain;
```

Comment:
nit: We can make this function a bit more concise and avoid some local variables like below:
```
std::string Filter::getDomain() { 
   const auto* specific_per_route_config =
      Http::Utility::resolveMostSpecificPerFilterConfig<FilterConfigPerRoute>(callbacks_);
   if (specific_per_route_config != nullptr && !specific_per_route_config->domain().empty()) {
      return specific_per_route_config->domain();
   }
   return config_->domain();
}
```